precompiles/dapp-staking-v3/README.md

```diff
@@ -0,0 +1,36 @@
+# dApp Staking Precompile Interface
+
+dApp Staking is at the core of Astar Network, an unique protocol used to incentivize builders to build
+great dApps on Astar.
+
+In order to improve it, the feature has undergone several overhauls in past.
+This is necessary to keep Astar competitive, and it's likely it will happen again in the future.
+
+Developers should account for this when developing dApps.
+Even though the interface compatibility will be kept as much as possible,
+as feature is changed & evolved, new functionality will become available, and
+interface will have to be expanded.
+
+The logic controlling dApp staking should be upgradable therefore.
+
+## V1 Interface
+
+**ONLY RELEVANT FOR DEVELOPERS WHO USED THE OLD INTERFACE.**
+
+Covers the _so-called_ `dApps staking v2` interface.
+
+Many actions that are done here as a part of a single call are broken down into multiple calls in the `v3` interface.
+Best effort is made to keep the new behavior as compatible as possible with the old behavior.
+Regardless, in some cases it's simply not possible.
+
+Some examples of this:
+* Since all stakes are reset at the end of each period, developers will need to adapt their smart contract logic for this.
+* Bonus rewards concept was only introduced from the precompile v2 interface, so there's no equivalent call in v1 interface.
```

Comment:
I know, but it's kind of confusing with the v1,v2 versioning for precompile and v2,v3 for pallet 😅

Reply:
How about I rename the `.sol` interfaces to v2 and v3?

Reply:
That would make more sense and less confusing.

Reply:
Should be done.